src/Compilers/Core/Portable/SourceGeneration/ISourceGenerator.cs

```diff
@@ -19,6 +19,7 @@ namespace Microsoft.CodeAnalysis
     /// is no guarantee that the same instance will be used on a 
     /// subsequent generation pass.
     /// </remarks>
+    [RoslynObsolete(Constants.UseIncrementalGenerator, DiagnosticId = Constants.UseIncrementalGeneratorDiagnosticId)]
```

Comment:
It seems like this would be a fairly noisy block -- because ISourceGenerator is also used as the "currency" for anything running generators, including incremental generators, since we wrap them. As long as any _implementor_ of ISourceGenerator is getting flagged via any of the other obsoletions, then I'd remove this to avoid a bunch of unnecessary noise.